specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,538 @@
+        amounts: Vec<U128>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+}
+    1. calls `compound::mt_on_transfer({"sender_id": "alice", "token_ids":["g133"], "amounts": ["1000000000000000000000"], msg: "interest-building"})`.
+    2. `compound` resolves the request/fails and `games` contract handles response from the promise with `games::mt_resolve_transfer` returning refunded amount if there is any or handling follow up from the result of compound cross contract call
+
+#### Batch Token deposit to a contract 
+
+Alice wants to deposit `silver` and `gold` tokens and the nft `gem` to a compound interest contract to earn some rewards. 
+
+Let's assume the following:
+- The `gold` token is represented  by the `games` contract with token_id `g133` .
+- The `silver` token is represented  by the `games` contract with token_id `s133` .
+- The `gem` unique only one nft token is represented  by the `games` contract with token_id `uu2` .
+- Alice's account is `alice`.
+- The compound interest contract is `compound`.
+- The precision on `gold` token is `10^18`. 
+- The precision on `silver` token is `10^18`. 
+- The 1000 tokens is `1000 * 10^18` or as a number is `1000000000000000000000`. 
+- The compound contract can work with many different token contracts and types.
+
+High-level explanation:
+
+Alice needs to issue a single transaction to `games` that will internally issue a cross contract call to `compound`. 
+
+The initial transaction to `games` is made with `compound` as the receiver of a set token_ids and amounts from `alice`.
+
+This call then waits on a response from `compound`. If `compound` responds with failure, the tx is aborted. 
+
+Otherwise `games` contract accepts the results and resolves the promise completing the transaction.
+
+- If transfer succeeded, `compound` can increase local ownership for `alice` to 1000 for `gold` token_id `g133`
+
+- If transfer fails, `compound` doesn't need to do anything in current example, but maybe can notify `alice` of unsuccessful transfer.
+
+Technical calls:
+1. `alice` calls `games::mt_transfer_batch_call({"receiver_id": "compound", amounts: ["1000000000000000000000","1000000000000000000000", "1"], "token_ids": ["g133","s133","uu2"], msg: "interest-building"})`.
+   During the `mt_transfer_call` call, `compound` does the following:
+     fn mt_on_transfer(
+        &mut self,
+        sender_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+}
+    1. calls `compound::mt_on_transfer({"sender_id": "alice", amounts: ["1000000000000000000000","1000000000000000000000", "1"], "token_ids": ["g133","s133","uu2"], msg: "interest-building"})`
+    2. `compound` resolves the request/fails and `games` contract handles response from the promise with `games::mt_resolve_transfer` returning refunded amount if there is any or handling follow up from the result of compound cross contract call
+
```

Comment:
How about take single transfer as a special case of batch transfer, and unify the interface into two: simple_transfer and transfer_call?

Reply:
if there's no pushback I'd be ok with this, I think the only reason to not do it that I can think of is if there's some gas saving operation that can be obtained from not having a single interface that just supports multiple token-ids

Reply:
Offline we discussed and came to the conclusion it'll be better to keep the functions separate.